flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/OverAggregateITCase.scala

```diff
@@ -532,6 +532,33 @@ class OverAggregateITCase extends BatchTestBase {
     )
   }
 
+  @Test
+  def testWindowAggregationSumWithQualify(): Unit = {
+    checkResult(
+      "SELECT d, e FROM Table5 QUALIFY sum(e) OVER (PARTITION BY d ORDER BY e) > 20",
```

Comment:
This is awesome. It will simplify stream deduplication. Can you also add a test for this and update the docs?
From:
```
SELECT [column_list]
FROM (
   SELECT [column_list],
     ROW_NUMBER() OVER ([PARTITION BY col1[, col2...]]
       ORDER BY time_attr [asc|desc]) AS rownum
   FROM table_name)
WHERE rownum = 1
```

To:
```
SELECT [column_list],
  ROW_NUMBER() OVER ([PARTITION BY col1[, col2...]] ORDER BY time_attr [asc|desc]) AS rownum
  FROM table_name)
  QUALIFY rownum = 1
```

Reply:
https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/deduplication/#deduplication

Reply:
added a test and changed the doc

one thing to clarify:
there are 2 options 
1. the one you mentioned above
```sql
SELECT [column_list],
    ROW_NUMBER() OVER ([PARTITION BY col1[, col2...]] ORDER BY time_attr [asc|desc]) AS rownum
FROM table_name
QUALIFY rownum = 1
```

2. the second one is slightly different
```sql
SELECT [column_list]
FROM table_name
 QUALIFY  ROW_NUMBER() OVER ([PARTITION BY col1[, col2...]] ORDER BY time_attr [asc|desc]) = 1
```  

From one side we can use alias in first option from another side there will be extra columns in projections